packages/suite/src/support/suite/FindBar/useFindInPage.ts

```diff
@@ -0,0 +1,268 @@
+            characterData: true,
+            attributes: false,
+        });
+    }, [rootElement]);
+
+    const cancelScheduled = useCallback(() => {
+        if (debounceTimer.current) {
+            clearTimeout(debounceTimer.current);
+            debounceTimer.current = null;
+        }
+    }, []);
+
+    const withObserverPaused = useCallback(
+        (fn: () => void) => {
+            const obs = observerRef.current;
+            isMutatingRef.current = true;
+            obs?.disconnect();
+            try {
+                fn();
+            } finally {
+                requestAnimationFrame(() => {
+                    isMutatingRef.current = false;
+                    obs?.takeRecords();
+                    observeRoot();
+                });
+            }
+        },
+        [observeRoot],
+    );
+
+    const queryMarkByOrdinal = (ord: number) =>
+        document.querySelector<HTMLElement>(
+            `${MARK_HIGHLIGHT_SELECTOR}[data-find-ordinal="${ord}"]`,
+        );
+
+    const applyActiveOrdinal = useCallback((ord: number | null, scrollIntoView = false) => {
+        const marks = Array.from(document.querySelectorAll<HTMLElement>(MARK_HIGHLIGHT_SELECTOR));
+        marks.forEach(m => m.setAttribute('data-active', 'false'));
+
+        if (ord == null) {
+            activeOrdinalRef.current = null;
+            setPosition(0);
+
+            return;
+        }
+
+        const el = queryMarkByOrdinal(ord);
+        if (!el) {
+            activeOrdinalRef.current = null;
+            setPosition(0);
+
+            return;
+        }
+
+        el.setAttribute('data-active', 'true');
+        if (scrollIntoView) el.scrollIntoView({ behavior: 'smooth', block: 'center' });
+
+        activeOrdinalRef.current = ord;
+        setPosition(ord + 1);
+    }, []);
+
+    const animateActivePulse = useCallback((ord: number) => {
+        const el = queryMarkByOrdinal(ord);
+        if (!el) return;
+
+        const old = el.querySelector(MARK_HIGHLIGHT_PULSE_SELECTOR);
+        if (old) old.remove();
+
+        const pulse = document.createElement('span');
+        pulse.className = MARK_HIGHLIGHT_PULSE_CLASSNAME;
+        el.appendChild(pulse);
+
+        pulse.animate(
+            [
+                { transform: 'scale(1.4)', opacity: 0.8 },
+                { transform: 'scale(1.0)', opacity: 0 },
+            ],
+            {
+                duration: 350,
+                easing: 'ease-out',
+                fill: 'forwards',
+            },
+        );
+
+        setTimeout(() => pulse.remove(), 400);
+    }, []);
+
+    const clampOrdinal = (ord: number, total: number) => {
+        if (total <= 0) return null;
+        if (ord < 0) return 0;
+        if (ord >= total) return total - 1;
+
+        return ord;
+    };
+
+    const removeAllMarks = useCallback(() => {
+        withObserverPaused(() => removeHighlights(rootElement));
+    }, [rootElement, withObserverPaused]);
+
+    const runHighlight = useCallback(
```

Comment:
I'd put here the comments explaining how it works if someone reads it he'd be surprised a lot.